### PR TITLE
noImplicitAny: Add types to reduce errors

### PR DESCRIPTION
**Pull request by tskarhed**
**What this PR does / why we need it**:
#14714 

**Special notes for your reviewer**:

Still need to write a test for the change in `TemplateSrv`. I have no idea how to work with Azure, so if someone could double check the change in `AzureMonitorQueryCtrl` to make sure it works, that would be great! It's a rather small change.

Turns out this issue has bit of a domino effect. I started with only editing 3 files. Also first PR since last year, so if there is any other feedback, please let me know!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "@types/react-transition-group": "2.0.16",
     "@types/react-virtualized": "9.18.12",
     "@types/react-window": "1.7.0",
+    "@types/remarkable": "^1.7.4",
```
Comment (jschill) [inline package.json]: Use fixed versions in package.json, ie 1.7.4 without the ^ - and don't forget to update the lock-file

File: scripts/ci-frontend-metrics.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 echo -e "Collecting code stats (typescript errors & more)"
 
-ERROR_COUNT_LIMIT=5150
+ERROR_COUNT_LIMIT=5120
```
Comment (jschill) [inline scripts/ci-frontend-metrics.sh]: 👍

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts [hunk 3]
```diff
@@ -360,7 +362,7 @@ export class AzureMonitorQueryCtrl extends QueryCtrl {
   getAutoInterval() {
     if (this.target.azureMonitor.timeGrain === 'auto') {
       return TimegrainConverter.findClosestTimeGrain(
-        this.templateSrv.builtIns.__interval.value,
+        this.templateSrv.getBuiltInIntervalValue,
```
Comment (jschill) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts]: Missing paranthesis?

**Review by jschill — CHANGES_REQUESTED** [review]

File: public/app/features/panel/panel_ctrl.ts [hunk 4]
```diff
@@ -99,7 +101,7 @@ export class PanelCtrl {
     }
   }
 
-  addEditorTab(title, directiveFn, index?, icon?) {
+  addEditorTab(title: string, directiveFn: any, index?: number, icon?: undefined) {
```
Comment (jschill) [inline public/app/features/panel/panel_ctrl.ts]: It looks like we're not using icon anymore. Should we remove it?

File: public/app/features/panel/panel_ctrl.ts [hunk 5]
```diff
@@ -36,7 +38,7 @@ export class PanelCtrl {
   timing: any;
   maxPanelsPerRowOptions: number[];
 
-  constructor($scope, $injector) {
+  constructor($scope: any, $injector: auto.IInjectorService) {
```
[the remark this answers is not in the document]
Comment (dehrax) [reply public/app/features/panel/panel_ctrl.ts]: Thanks! Couldn't find anything for `$scope`

File: public/app/features/panel/panel_ctrl.ts (hunk 4 above)
Comment (dehrax) [reply public/app/features/panel/panel_ctrl.ts]: @hugohaggmark Any input on this?

Comment (hugohaggmark) [reply public/app/features/panel/panel_ctrl.ts]: Maybe change to any?

```suggestion
  addEditorTab(title: string, directiveFn: any, index?: number, icon?: any) {
```

Comment (torkelo) [reply public/app/features/panel/panel_ctrl.ts]: yea, not used anymore I think. external plugins could use it I think, but not shown in UI anymore. but keep for now

**Review by torkelo — APPROVED** [review]
Looks good!

**Review by torkelo — APPROVED** [review]

**Pull request merged**